### ISSUE TITLE
master breaks in Atom context

### ISSUE THREAD

**Issue by gre** [eslint/eslint#3163]
Following issue https://github.com/eslint/eslint/issues/3156 , the current master does not run in Atom context's because of the use of `is-json-valid` at https://github.com/eslint/eslint/blob/master/lib/testers/eslint-tester.js#L74

The error is following:

```
Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".
```

caused by `/Users/gre/dev/eslint/node_modules/is-my-json-valid/node_modules/generate-function/index.js:55`

So if you want to preserve this code, I suggest that `var validateSchema = validate(metaSchema, { verbose: true });` would be computed lazily instead (at the first `ESLintTester` instanciation).

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
As far as I know, in Atom, a special way is needed to use `Function` constructor. (see also https://github.com/AtomLinter/linter-eslint/issues/99#issuecomment-112214088)

`eslint-tester` is loaded at https://github.com/eslint/eslint/blob/eaa37c265bc407afa4944e49bbbfac6de770def0/lib/api.js#L12.
And as @gre pointed out, `Function` constructor is used in  https://github.com/eslint/eslint/blob/eaa37c265bc407afa4944e49bbbfac6de770def0/lib/testers/eslint-tester.js#L74

I'm not sure, it may mean `require("eslint")` needs a special way.

```js
var allowUnsafeNewFunction = require("loophole").allowUnsafeNewFunction;
var eslint;
allowUnsafeNewFunction(function() {
    eslint = require("eslint");
});
...
```

If my conjecture is correct, It's inconvenience.

**Comment by nzakas** [comment]
@gre can you explain more? I'm having a hard time understanding what's going on.

**Comment by ide** [comment]
Atom puts some restrictions on eval with CSP and one of eslint's dependencies (namely is-my-json-valid -> generate-function) calls `Function.apply` which breaks the Atom linter. I'll see if that loophole module helps.

**Comment by zyklus** [comment]
+1

**Comment by jimCresswell** [comment]
+1 thanks @ide

**Comment by ide** [comment]
I got it working but needed to patch loophole, which I'm waiting on.

**Comment by jimCresswell** [comment]
Much appreciated, I wonder how many semicolons I've lost while it's been broken :smile:

**Comment by IanVS** [comment]
Great, can someone confirm this is fixed in AtomLinter, and this issue can be closed?

**Comment by jimCresswell** [comment]
I'm waiting for the linter-eslint project to release an update with the changes so I can check in that context  https://github.com/AtomLinter/linter-eslint/releases

**Comment by ide** [comment]
I think this issue can be closed as a wontfix since linter-eslint can work around it.

**Comment by nzakas** [comment]
Sounds good
